kokoro/config/test/image_lists.gcl

```diff
@@ -15,149 +15,149 @@ local template _distro {
 }
 focal_aarch64 = _distro {
   release = [
-      'ubuntu-2004-lts-arm64',
-      'ubuntu-minimal-2004-lts-arm64',
+      'ubuntu-os-cloud:ubuntu-2004-lts-arm64',
+      'ubuntu-os-cloud:ubuntu-minimal-2004-lts-arm64',
   ]
   presubmit = [
-      'ubuntu-2004-lts-arm64',
+      'ubuntu-os-cloud:ubuntu-2004-lts-arm64',
   ]
 }
 jammy_x86_64 = _distro {
   release = [
-    'ubuntu-2204-lts',
-    'ubuntu-minimal-2204-lts',
+    'ubuntu-os-cloud:ubuntu-2204-lts',
+    'ubuntu-os-cloud:ubuntu-minimal-2204-lts',
   ]
   presubmit = [
-      'ubuntu-minimal-2204-lts',
+      'ubuntu-os-cloud:ubuntu-minimal-2204-lts',
   ]
 }
 jammy_aarch64 = _distro {
   release = [
-    'ubuntu-2204-lts-arm64',
-    'ubuntu-minimal-2204-lts-arm64',
+    'ubuntu-os-cloud:ubuntu-2204-lts-arm64',
+    'ubuntu-os-cloud:ubuntu-minimal-2204-lts-arm64',
   ]
   presubmit = [
-      'ubuntu-minimal-2204-lts-arm64',
+      'ubuntu-os-cloud:ubuntu-minimal-2204-lts-arm64',
   ]
 }
 
 // RPM Linux distros. (Do not modify this comment.)
 centos7_x86_64 = _distro {
   release = [
     // CentOS.
-    'centos-7',
+    'centos-cloud:centos-7',
     // RHEL.
-    'rhel-7',
-    'rhel-7-9-sap-ha',
+    'rhel-cloud:rhel-7',
+    'rhel-sap-cloud:rhel-7-9-sap-ha',
   ]
-  presubmit = ['centos-7']
+  presubmit = ['centos-cloud:centos-7']
 }
 centos8_x86_64 = _distro {
   release = [
     // RHEL.
-    'rhel-8',
-    'rhel-8-2-sap-ha',
-    'rhel-8-4-sap-ha',
-    'rhel-8-6-sap-ha',
+    'rhel-cloud:rhel-8',
+    'rhel-sap-cloud:rhel-8-2-sap-ha',
+    'rhel-sap-cloud:rhel-8-4-sap-ha',
+    'rhel-sap-cloud:rhel-8-6-sap-ha',
     // Rocky.
-    'rocky-linux-8',
+    'rocky-linux-cloud:rocky-linux-8',
   ]
-  presubmit = ['rocky-linux-8']
+  presubmit = ['rocky-linux-cloud:rocky-linux-8']
 }
 rockylinux9_x86_64 = _distro {
   release = [
     // RHEL.
-    'rhel-9',
+    'rhel-cloud:rhel-9',
     // Rocky.
-    'rocky-linux-9',
+    'rocky-linux-cloud:rocky-linux-9',
   ]
-  presubmit = ['rocky-linux-9']
+  presubmit = ['rocky-linux-cloud:rocky-linux-9']
 }
 rockylinux9_aarch64 = _distro {
   release = [
     // RHEL.
-    'rhel-9-arm64',
+    'rhel-cloud:rhel-9-arm64',
     // Rocky.
-    'rocky-linux-9-arm64',
+    'rocky-linux-cloud:rocky-linux-9-arm64',
   ]
-  presubmit = ['rocky-linux-9-arm64']
+  presubmit = ['rocky-linux-cloud:rocky-linux-9-arm64']
 }
 sles12_x86_64 = _distro {
   release = [
-    'sles-12',
-    'sles-12-sp5-sap',
+    'suse-cloud:sles-12',
+    'suse-sap-cloud:sles-12-sp5-sap',
   ]
-  presubmit = ['sles-12']
+  presubmit = ['suse-cloud:sles-12']
 }
 sles15_x86_64 = _distro {
   release = [
-    'sles-15',
-    'sles-15-sp2-sap',
-    'sles-15-sp5-sap',
-    'opensuse-leap',
-    'opensuse-leap-15-4',
+    'suse-cloud:sles-15',
+    'suse-sap-cloud:rhel-sap-cloud:sles-15-sp2-sap',
+    'suse-sap-cloud:rhel-sap-cloud:sles-15-sp5-sap',
+    'opensuse-cloud:opensuse-leap',
+    'opensuse-cloud=opensuse-leap-15-4-v20231208-x86-64',
```

Comment:
IMO, this exposes unnecessary implementation details. The key here is that we're testing on `opensuse-leap-15-4` — the specific image is somewhat irrelevant. I wish GCE exposed this as a separate image family, but it doesn't, so we invented one. We might still want to override a specific image (family), even for one of the existing ones (e.g., to temporarily pin the image if a broken one is rolled out) without changing the intent of the tests.

Reply:
This doesn't expose anything we aren't already exposing?

Reply:
I agree that it's not a great situation to be in, but I think it's far better to specify the image in one place, rather than two.

Reply:
Initially, this provided a layer of abstraction that allowed separating the concerns of "what we intend to test" vs "what that actually translates to". Conflating them could make the former harder to discover. That said, having the override in the same file _is_ nicer, so maybe the right solution here is to rely on a comment instead…